frontend/app/app.js

```diff
@@ -0,0 +1,122 @@
+var app = angular.module('movies', []);
+var API = "http://localhost:8080/api/"; //Your server url goes here
+
+/**
+ * Here we define a new controller called MoviesController. 
+ * We require $http, so angular injects it automatically. 
+ */
+app.controller("MoviesController", ['$http', function($http) {
+
+	var collection = this; //Save this controller 
+	/**
+	 * Use the $http method to send a GET request to our movie API. 
+	 * We provide some details in the first paramater. 
+	 * We use the .then() method to pass in our success and error functions. 
+	 */
+	$http({
+	 method: 'GET',
+	 url: API+'movies',
+	 headers: {
+	   'Content-Type': undefined
+	 }
+	}).then(function success(response) {
+		
+		//If this was successfull, handle the data from the server
+		collection.movies = response.data; //Copy the response data into our movie array
+
+		//Load the image for each movie
+		angular.forEach(collection.movies, function(movie) {
+			fetchMovieImage(movie);
+		})
+
+	}, function error(error){
+		console.log(error);
+	});
+
+	/**
+	 * Adds a new movie to the array.
+	 */
+	collection.addNewMovie = function() {
+		/**
+		 * Like before, we use the $http method to send a POST request to our movie API. 
+		 * Because its a POST request we provide the data we want to send.
+		 * We use the .then() method to pass in our success and error functions. 
+		 */
+		$http({
+			method: 'POST',
+			url: API+'movies',
+			data: {
+				title: collection.newMovie.title,
+				description: collection.newMovie.description
+			}
+		}).then(function successCallback(response) {
+			//Make sure the response has the data we need
+			if(response.data.title != null && response.data.description != null) {
+				//Log the data just for reference
+				console.log(JSON.stringify(response.data)); //We use JSON.stringify to make the response data human-readable
+				collection.newMovie = {}; //Clear the newMovie data to reset the form. 
+			    collection.movies.push(response.data); //Add the new movie to our movie array
+			    collection.expandNewMovieEntry = false; //Close the movie creation form
+			    fetchMovieImage(response.data); //Load the image for this new movie
+			} else {
+				//The response was missing data, something went wrong. 
+				console.log("Failed: "+JSON.stringify(response.data)); //We use JSON.stringify to make the response data human-readable
```

Comment:
Maybe actually display on error on page instead of failing silently in the console.